### PR TITLE
More EnC tests for closures

### PR DESCRIPTION
**Pull request by ivanbasov**

### CODE REVIEW

**Comment by ivanbasov** [comment]
@dotnet-bot  test please

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs [hunk 1]
```diff
@@ -3631,5 +3631,407 @@ .maxstack  9
...
+            var f1 = compilation1.GetMember<MethodSymbol>("C.F");
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(new SemanticEdit(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            var md1 = diff1.GetMetadata();
+            var reader1 = md1.Reader;
```
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: >ar reader1 = md1.Reader; [](start = 13, length = 24)

We should validate synthesized members using VerifySynthesizedMembers

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs [hunk 2]
```diff
@@ -3631,5 +3631,407 @@ .maxstack  9
...
+            var source1 = MarkedSource(@"
+public class C 
+    {
+    public void F()            
+    {
+        int x = 0;
+        {
+            int y = 0;
+            // Captures two struct closures
+            int L() => x + y + 1;
```
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: >int L() => x + y + 1; [](start = 12, length = 21)

This is not testing what we want - you are missing <N> markers to let the compiler know what syntax nodes match

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs [hunk 3]
```diff
@@ -3631,5 +3631,407 @@ .maxstack  9
...
+                ImmutableArray.Create(new SemanticEdit(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            var md1 = diff1.GetMetadata();
+            var reader1 = md1.Reader;
+
+            CheckEncLogDefinitions(reader1,
+                Row(2, TableIndex.StandAloneSig, EditAndContinueOperation.Default),
+                Row(5, TableIndex.TypeDef, EditAndContinueOperation.Default),
+                Row(6, TableIndex.TypeDef, EditAndContinueOperation.Default),
+                Row(5, TableIndex.TypeDef, EditAndContinueOperation.AddField),
```
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: >.AddField [](start = 67, length = 9)

There should be no additions

Comment (ivanbasov) [reply src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: Ouch, yes. Sorry! Thanks!

---
In reply to: [137682150](https://github.com/dotnet/roslyn/pull/21971#discussion_r137682150) [](ancestors = 137682150)

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs (hunk 2 above)
Comment (ivanbasov) [reply src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: Thanks!

---
In reply to: [137682130](https://github.com/dotnet/roslyn/pull/21971#discussion_r137682130) [](ancestors = 137682130)

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs (hunk 1 above)
Comment (ivanbasov) [reply src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs]: Thank you!

---
In reply to: [137682070](https://github.com/dotnet/roslyn/pull/21971#discussion_r137682070) [](ancestors = 137682070)

**Comment by ivanbasov** [comment]
test perf_correctness_prtest

**Pull request merged**